source/Symbol/SwiftASTContext.cpp

```diff
@@ -3472,9 +3472,16 @@ bool SwiftASTContext::LoadOneImage(Process &process, FileSpec &link_lib_spec,
 
 static std::vector<std::string>
 GetLibrarySearchPaths(const swift::SearchPathOptions &search_path_opts) {
-  std::vector<std::string> paths(search_path_opts.LibrarySearchPaths.begin(),
-                                 search_path_opts.LibrarySearchPaths.end());
+  // The order in which we look up the libraries is important. The REPL
+  // dlopen()s libswiftCore, and gives precedence to the just built standard
+  // library instead of the one in the OS. When we type `import Foundation`,
+  // we want to make sure we end up loading the correct library, i.e. the
+  // one sitting next to the stdlib we just built, and then fall back to the
+  // one in the OS if that's not available.
+  std::vector<std::string> paths;
```

Comment:
(Nit) We might want to reserve the vector.

Reply:
This is really a micro-optimization, as this codepath is called only once per dylib loaded. I looked what's the typical # of times this is called on some projects, and it's ~100 (ballpark).
Also, the number of elements we end up having in the array is almost always 2 (one search path is `/usr/lib/swift`, and the other is the `stdlib` you just built). 

That said, your comment made me realize that we should only push the path if it's not an empty string. I guess I'll make that change.